catalog/core/catalog-core-standardframework/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -72,24 +72,59 @@
         <argument ref="catalogFramework"/>
     </bean>
 
-    <bean id="sourcePoller" class="ddf.catalog.util.impl.SourcePoller">
-        <cm:managed-properties persistent-id="ddf.catalog.util.impl.SourcePoller"
+    <service ref="deprecatedProductCache" interface="ddf.catalog.cache.ResourceCacheInterface"/>
+
+    <bean id="sourceStatusSourcePoller" class="ddf.catalog.util.impl.SourcePollerFactory"
+          factory-method="createSourceStatusSourcePoller"/>
+    <bean id="sourceStatusSourcePollerRunner" class="ddf.catalog.util.impl.SourcePollerRunner"
+          destroy-method="destroy">
+        <argument ref="sourceStatusSourcePoller"/>
+        <argument>
+            <bean class="java.util.concurrent.Executors"
+                  factory-method="newSingleThreadScheduledExecutor">
+                <argument>
+                    <bean class="org.codice.ddf.platform.util.StandardThreadFactoryBuilder"
+                          factory-method="newThreadFactory">
+                        <argument value="sourceStatusSourcePollerRunnerThread"/>
+                    </bean>
+                </argument>
+            </bean>
+        </argument>
+        <!--TODO How can we guarantee that the sources have been set before starting the polling?-->
```

Comment:
❓

Reply:
You could kick off the polling thread in an init-method which won't be called until all properties have been set.